allensdk/brain_observatory/behavior/write_nwb/__main__.py

```diff
@@ -21,15 +21,31 @@ def write_behavior_ophys_nwb(session_data, nwb_filepath):
     nwb_filepath_error = nwb_filepath+'.error'
 
     # Clean out files from previous runs:
-    for filename in [nwb_filepath_inprogress, nwb_filepath_error, nwb_filepath]:
+    for filename in [nwb_filepath_inprogress,
+                     nwb_filepath_error,
+                     nwb_filepath]:
         if os.path.exists(filename):
             os.remove(filename)
 
     try:
-        session = BehaviorOphysSession(api=BehaviorOphysJsonApi(session_data))
-        BehaviorOphysNwbApi(nwb_filepath_inprogress).save(session)
-        api = BehaviorOphysNwbApi(nwb_filepath_inprogress)
-        assert sessions_are_equal(session, BehaviorOphysSession(api=api))
+        json_session = BehaviorOphysSession(
```

Comment:
This logic is nearly identical to the validation logic in write_behavior_nwb. Any way to encapsulate into a single method?